src/SharpGLTF.Toolkit/Geometry/Packed/PackedPrimitiveBuilder.cs

```diff
@@ -93,32 +93,57 @@ public void SetMorphTargets(IPrimitiveReader<TMaterial> srcPrim, PackedEncoding
             bool hasNormals = _VertexAccessors.Any(item => item.Attribute.Name == "NORMAL");
             bool hasTangents = _VertexAccessors.Any(item => item.Attribute.Name == "TANGENT");
 
+            bool hasColors0 = _VertexAccessors.Any(item => item.Attribute.Name == "COLOR_0");
+            bool hasColors1 = _VertexAccessors.Any(item => item.Attribute.Name == "COLOR_1");
+            bool hasTextCoords0 = _VertexAccessors.Any(item => item.Attribute.Name == "TEXCOORD_0");
+            bool hasTextCoords1 = _VertexAccessors.Any(item => item.Attribute.Name == "TEXCOORD_1");
+
             if (!hasPositions) throw new InvalidOperationException("Set vertices before morph targets.");
 
             for (int i = 0; i < srcPrim.MorphTargets.Count; ++i)
             {
                 var mtv = srcPrim.MorphTargets[i].GetMorphTargetVertices(srcPrim.Vertices.Count);
 
                 var pAccessor = VertexTypes.VertexUtils.CreateVertexMemoryAccessor(mtv, "POSITIONDELTA", vertexEncodings);
+                //TODO: Should POSITIONDELTA be removed if all delta is 0s?

```

Comment:
yes, it makes sense to remove POSITIONDELTA now, becase we're going to remove the previous hasPositions check.

Technically, there's a dependency between position, normals and tangents. But tangents also depend on UV coordinates. So it is possible that if we apply a morph over UV coordinates, tangents might also be required to be morphed....

So it makes sense that every vertex channel to be independent from the other

Reply:
done